src/fides/api/models/sql_models.py

```diff
@@ -326,6 +328,36 @@ class PolicyCtl(Base, FidesBase):
     rules = Column(JSON)
 
 
+def create_data_categories_property(dataset_references):
+    """
+    Creates a column property that extracts a unique set of data categories from multiple datasets.
+
+    Takes a list of dataset references and returns a column property containing a single array of
+    all unique data categories found across all the specified datasets. This combines and deduplicates
+    data categories from every level of the JSON structure in each dataset's collections.
+    """
+
+    subquery = (
+        select(
+            [
+                func.jsonb_array_elements_text(
+                    text(
+                        "jsonb_path_query(collections::jsonb, '$.** ? (@.data_categories != null).data_categories')"
```

Comment:
This is the key change, the JSONB path returns all the `data_categories` at all levels of nesting. The `create_data_categories_property` function is used to create the `dataset_data_categories` column property for `System` and `PrivacyDeclaration`